src/party_context_menu.c

```diff
@@ -0,0 +1,1290 @@
+    FillBgTilemapRect(partyMenu->bgConfig, GF_BG_LYR_MAIN_0, tiles[5], rect[0] + rect[2] - 1, rect[1] + 1, 1, rect[3] - 2, TILEMAP_FILL_OVWT_PAL);
+    FillBgTilemapRect(partyMenu->bgConfig, GF_BG_LYR_MAIN_0, tiles[6], rect[0] + 1, rect[1], rect[2] - 2, 1, TILEMAP_FILL_OVWT_PAL);
+    FillBgTilemapRect(partyMenu->bgConfig, GF_BG_LYR_MAIN_0, tiles[7], rect[0] + 1, rect[1] + rect[3] - 1, rect[2] - 2, 1, TILEMAP_FILL_OVWT_PAL);
+}
+
+void sub_0207E54C(PartyMenuStruct *partyMenu, int numItems, int selection, int state) {
+    for (int i = 0; i < numItems; ++i) {
+        if (selection == i) {
+            sub_0207E3A8(partyMenu, numItems, i, state, 1);
+        } else {
+            sub_0207E3A8(partyMenu, numItems, i, state, 0);
+        }
+    }
+}
+
+PartyMenuContextMenuCursor *PartyMenu_CreateContextMenuCursor(PartyMenuStruct *partyMenu, const PartyMenuContextMenu *template, int selection, HeapID heapId, int state) {
+    PartyMenuContextMenuCursor *ret = AllocFromHeap(heapId, sizeof(PartyMenuContextMenuCursor));
+    ret->menu = *template;
+    ret->numItems = ret->menu.numItems;
+    ret->prevSelection = selection;
+    ret->selection = selection;
+    ret->state = state;
+    PartyMenu_ShowContextMenu(partyMenu, ret->menu.numItems, ret->state);
+    sub_0207E358(partyMenu, &ret->menu, ret->menu.numItems, ret->prevSelection, ret->state);
+    PartyMenu_SetBlendBrightness_ForYesNo();
+    Set2dSpriteVisibleFlag(partyMenu->sprites[PARTY_MENU_SPRITE_ID_9], FALSE);
+    if (partyMenu->args->context == PARTY_MENU_CONTEXT_UNION_ROOM_BATTLE_SELECT || partyMenu->args->context == PARTY_MENU_CONTEXT_17 || partyMenu->args->context == PARTY_MENU_CONTEXT_BATTLE_HALL || partyMenu->args->context == PARTY_MENU_CONTEXT_23) {
+        Set2dSpriteVisibleFlag(partyMenu->sprites[PARTY_MENU_SPRITE_ID_8], FALSE);
+    }
+    return ret;
+}
+
+void PartyMenu_CloseContextMenu(PartyMenuStruct *partyMenu, PartyMenuContextMenuCursor *cursor) {
+    PartyMenu_HideContextMenu(partyMenu, cursor->numItems, cursor->state);
+    FreeToHeap(cursor);
+    if (partyMenu->args->context == PARTY_MENU_CONTEXT_4 || partyMenu->args->context == PARTY_MENU_CONTEXT_SPIN_TRADE) {
+        Set2dSpriteVisibleFlag(partyMenu->sprites[PARTY_MENU_SPRITE_ID_9], FALSE);
+    } else {
+        Set2dSpriteVisibleFlag(partyMenu->sprites[PARTY_MENU_SPRITE_ID_9], TRUE);
+    }
+    if (partyMenu->args->context == PARTY_MENU_CONTEXT_UNION_ROOM_BATTLE_SELECT || partyMenu->args->context == PARTY_MENU_CONTEXT_17 || partyMenu->args->context == PARTY_MENU_CONTEXT_BATTLE_HALL || partyMenu->args->context == PARTY_MENU_CONTEXT_23) {
+        Set2dSpriteVisibleFlag(partyMenu->sprites[PARTY_MENU_SPRITE_ID_8], TRUE);
+    }
+}
+
+// NOTE: naming style reflective of that these do not operate on the central type `PartyMenuStruct *` of this TU
+static BOOL hitboxHasSelectionMapping(int state, int numItems, int selection) {
```

Comment:
still stay consistent with other function naming styles and keep the first letters capitalized

Reply:
i'd rather have this visual distinction for these kinds of helpers